docs/neural_modeling/plot_05_place_cells.py

```diff
@@ -0,0 +1,410 @@
+
+X = basis.evaluate(position, theta, speed)
+
+# %%
+# `X` is our design matrix. For each timestamps, it contains the information about the current position, speed and theta phase of the experiment. Notice how passing a pynapple object to `evaluate` also returns a `pynapple` object.
+
+print(X)
+
+# %%
+# ## Model learning
+#
+# We can now use the Poisson GLM from nemos to learn the model.
+
+glm = nmo.glm.GLM(
+    regularizer=nmo.regularizer.UnRegularized("LBFGS", solver_kwargs=dict(tol=10**-12))
+)
+
+glm.fit(X[:, np.newaxis, :], count[:, np.newaxis])
+
+# %%
+# ## Prediction
+#
+# Let's check first if our model can accurately predict the different tuning curves we displayed above. We can use the `predict` function of nemos and then compute new tuning curves
+
+predicted_rate = glm.predict(X[:, np.newaxis, :]) / bin_size
+
+glm_pf = nap.compute_1d_tuning_curves_continuous(predicted_rate, position, 50)
+glm_pos_theta, xybins = nap.compute_2d_tuning_curves_continuous(
+    predicted_rate, data, 30
+)
+glm_speed = nap.compute_1d_tuning_curves_continuous(predicted_rate, speed, 30)
+
+# %%
+# Let's display both tuning curves together.
+
+plt.figure()
+gs = plt.GridSpec(2, 2)
+plt.subplot(gs[0, 0])
+plt.plot(pf[neuron])
+plt.plot(glm_pf[0], label="GLM")
+plt.xlabel("Position (cm)")
+plt.ylabel("Firing rate (Hz)")
+plt.legend()
+
+plt.subplot(gs[0, 1])
+plt.plot(tc_speed[neuron])
+plt.plot(glm_speed[0], label="GLM")
+plt.xlabel("Speed (cm/s)")
+plt.ylabel("Firing rate (Hz)")
+plt.legend()
+
+plt.subplot(gs[1, 0])
+extent = (xybins[0][0], xybins[0][-1], xybins[1][0], xybins[1][-1])
+plt.imshow(tc_pos_theta[neuron].T, aspect="auto", origin="lower", extent=extent)
+plt.xlabel("Position (cm)")
+plt.ylabel("Theta Phase (rad)")
+
+plt.subplot(gs[1, 1])
+plt.imshow(glm_pos_theta[0].T, aspect="auto", origin="lower", extent=extent)
+plt.xlabel("Position (cm)")
+plt.ylabel("Theta Phase (rad)")
+plt.title("GLM")
+
+plt.tight_layout()
+
+# %%
+# ## Model selection
+#
+# While this model captures nicely the features-rate relationship, it is not necessarily the simplest model. Let's construct several models and evaluate their score to determine the best model.
+#
+# !!! note "To shorten this notebook, only a few combination are tested. Feel free to expand this list."
+#
+
+models = {
+    "position": position_basis,
+    "position + speed": position_basis + speed_basis,
+    "position + phase": position_basis + phase_basis,
+    "position + phase + speed": position_basis + phase_basis + speed_basis,
+}
+features = {
+    "position": (position,),
+    "position + speed": (position, speed),
+    "position + phase": (position, theta),
+    "position + phase + speed": (position, theta, speed),
+}
+
+# %%
+# In a loop, we can (1) evaluate the basis, (2), fit the model, (3) compute the score and (4) predict the firing rate. For evaluating the score, we can define a train set of intervals and a test set of intervals.
+
+train_iset = position.time_support[
```

Comment:
Since we have a bunch of disjoint epochs, we could do: position.time_support[::2] for train and position.time_support[1::2] as a better splitting strategy, so that we sample more "uniformily" in time.

Reply:
Done